lib/saml/user_attributes/dslogon.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+require 'saml/user_attributes/base_decorator'
+
+# TODO: remove these nocov comments when this is able to be tested.
+
+module SAML
+  module UserAttributes
+    class DSLogon < BaseDecorator
+      PREMIUM_LOAS = %w(2 3).freeze
+
+      def uuid
+        attributes['uuid']
+      end
+
+      def email
+        attributes['email']
+      end
+
+      def dslogon_edipi
+        attributes['dslogon_uuid']
+      end
+
+      def dslogon_status
+        attributes['dslogon_status']
+      end
+
+      def gender
+        attributes['dslogon_gender']&.chars&.first&.upcase
+      end
+
+      def dslogon_deceased
+        attributes['dslogon_deceased']
+      end
+
+      def birth_date
+        attributes['dslogon_birth_date']
+      end
+
+      def first_name
+        attributes['dslogon_fname']
+      end
+
+      def middle_name
+        attributes['dslogon_mname']
+      end
+
+      def last_name
+        attributes['dslogon_lname']
+      end
+
+      def dslogon_idtype
+        attributes['dslogon_idtype']
+      end
+
+      def ssn
+        attributes['dslogon_idvalue']
+      end
+
+      def dslogon_assurance
+        attributes['dslogon_assurance']
+      end
+
+      def loa
+        { current: loa_current, highest: loa_highest }
+      end
+
+      def multifactor
+        attributes['dslogon_multifactor']
```

Comment:
this should be `attributes['multifactor']` - see https://github.com/department-of-veterans-affairs/vets.gov-team/issues/4965